src/error.rs

```diff
@@ -1,18 +1,30 @@
-use thiserror::Error;
+use core::fmt;
 
 /// An error related to Ed25519 signatures.
-#[derive(Error, Debug, Copy, Clone, Eq, PartialEq)]
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
 pub enum Error {
     /// The encoding of a secret key was malformed.
-    #[error("Malformed secret key encoding.")]
     MalformedSecretKey,
     /// The encoding of a public key was malformed.
-    #[error("Malformed public key encoding.")]
     MalformedPublicKey,
     /// Signature verification failed.
-    #[error("Invalid signature.")]
     InvalidSignature,
     /// A byte slice of the wrong length was supplied during parsing.
-    #[error("Invalid length when parsing byte slice.")]
     InvalidSliceLength,
 }
+
+impl fmt::Display for Error {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let msg = match self {
+            Self::MalformedSecretKey => "Malformed secret key encoding.",
+            Self::MalformedPublicKey => "Malformed public key encoding.",
+            Self::InvalidSignature => "Invalid signature.",
+            Self::InvalidSliceLength => "Invalid length when parsing byte slice.",
+        };
+
+        msg.fmt(f)
+    }
+}
```

Comment:
Optional: Could be simplified a bit with `displaydoc` as described here https://github.com/dtolnay/thiserror/pull/64#issuecomment-735805334

Reply:
I can do it if you think that the extra dependency is worth it

Reply:
Let's merge this, we can clean this up later